### PR TITLE
All project components plus logic tested

### PR DESCRIPTION
**Pull request by MMhaidari**
# What is added new 
- [ ] Unit tests have been added for the logic file 
- [ ] Unit tests added for components 
- [ ] Snapshot testing used for test components 
- [ ] all tests works fine and pass

### CODE REVIEW

File: src/tests/operate.test.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+  test('adds two numbers', () => {
+    expect(operate('10', '5', '+')).toBe('15');
+    expect(operate('0.1', '0.2', '+')).toBe('0.3');
+  });
+
+  test('subtracts two numbers', () => {
+    expect(operate('10', '5', '-')).toBe('5');
+    expect(operate('-0.1', '-0.2', '-')).toBe('0.1');
+  });
```
Comment (elisha2kyakpo1) [inline src/tests/operate.test.js]: - [ ] Kindly add more tests, we are supposed to test all the operations and scenarios. For example, the multiplication and the modulus operators are not covered.

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @MMhaidari and @henry-dura, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ The PR title and description look good and professional.
✔️ No linter errors.
✔️ Descriptive readme file.
✔️ Tests passing👏

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by MMhaidari** [comment]
# What is added 
- [ ] tests added for multiplication and modulo operator as instructed 
- [ ] all operators  tests covered

**Review by Zuhaib042 — APPROVED** [review]
Hi @MMhaidari , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights
✅ Professional PR
✅ No linters errors
✅ Tests are passing
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**